### ISSUE TITLE
[Bug] Dangling output operator in graph moved as final graph outputs in fused graph.

### ISSUE THREAD

**Issue by TedThemistokleous** [microsoft/onnxruntime#26176]
### Describe the issue

Related to PR - https://github.com/microsoft/onnxruntime/pull/26092

We're running a customer model and have observed the following behavior:

Graph contains op nodes (batchnormalization) with outputs that aren't used in the Onnx model parsed in.

We've determined and seen that this is being added in GraphViewerToProto()

Netron output for example from the dumped model after we get to compile() but before MIGraphX does a compile on the resulting Onnx stream

<img width="1500" height="1057" alt="Image" src="https://github.com/user-attachments/assets/7971a7dc-6204-49b4-85c8-655a824cfbcc" />

Is there a way to prune these outputs so they're not being used in the Onnx stream or is this some interaction/thing we need to handle with ToProto() calls? 

We used log warnings to track this and got the following as well

<img width="1244" height="1709" alt="Image" src="https://github.com/user-attachments/assets/816eab6e-f7aa-4c3a-b0e4-def49df0d086" />

### To reproduce

Run a model with dangling outputs for batch normalization then attempt a compile in the EP

### Urgency

Customer model for us when using Onnxruntime. This unfortunately breaks the inference run even though the same model and same input in MIGraphX-driver doesn't yield any faults likely due to uninitialized outputs now that Onnxruntime is trying to write. 

### Platform

Linux

### OS Version

Ubuntu 24.04

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

Seen on mainline Onnxruntime as well. 

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

MIGraphX

### Execution Provider Library Version

ROCm 7.0

**Comment by TedThemistokleous** [comment]
Closing this out. Found solution. Path forward is here - > https://github.com/microsoft/onnxruntime/pull/26224
